### PR TITLE
Fix #23787

### LINKED ISSUE #23787: Cookie locale detection is broken if cookie name is similar with NEXT_COOKIE
### What version of Next.js are you using?

10.1.3

### What version of Node.js are you using?

12.14.0

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying your application?

next start

### Describe the Bug

I renamed cookie name from NEXT_COOKIE to NEXT_COOKIE2 and rendering is permanently the the following error:
```
message=Cannot read property 'toLowerCase' of undefined, stack=TypeError: Cannot read property 'toLowerCase' of undefined
    at find (<path>/node_modules/next/next-server/lib/i18n/detect-locale-cookie.ts:7:39)
    at Array.find (<anonymous>)
    at detectLocaleCookie (<path>/node_modules/next/next-server/lib/i18n/detect-locale-cookie.ts:6:20)
    at DevServer.requestHandler [as handleRequest] (<path>/node_modules/next/next-server/server/next-server.ts:436:28)
    at <path>/node_modules/next/server/next.ts:38:14
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

### Expected Behavior

I expect that all the similar names will be ignored and check will go to the step with accept-language header.

### To Reproduce

Just name some cooke NEXT_COOKIE2 or something that contain NEXT_COOKIE and any page will be failed.

### PR DESCRIPTION
**Pull request by timbset**
## Bug

- [x] Related issues linked using `fixes #number`
- [ ] Integration tests added

Fixes: #23787

### CODE REVIEW

**Review by ijjk — COMMENTED** [review]
Can we add a test case to the i18n test suite to ensure an error doesn't occur when a similar header is sent as mentioned?

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| buildDuration | 11.6s | 11.4s | -143ms |
| nodeModulesSize | 46 MB | 46 MB | ⚠️ +13 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.463 | 2.55 | ⚠️ +0.09 |
| / avg req/sec | 1015.14 | 980.22 | ⚠️ -34.92 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.449 | 1.653 | ⚠️ +0.2 |
| /error-in-render avg req/sec | 1724.88 | 1512.17 | ⚠️ -212.71 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| 677f882d2ed8..e6c6.js gzip | 13.4 kB | 13.4 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 7.12 kB | 7.12 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 60.2 kB | 60.2 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| _app-8fbabfc..6440.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-d07ca..ef4d.js gzip | 3.71 kB | 3.71 kB | ✓ |
| amp-9716187d..0aa8.js gzip | 536 B | 536 B | ✓ |
| hooks-107e90..74c7.js gzip | 888 B | 888 B | ✓ |
| index-ac435c..ecf2.js gzip | 227 B | 227 B | ✓ |
| link-c31053f..c329.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..dc9d.js gzip | 303 B | 303 B | ✓ |
| withRouter-6..0e02.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.89 kB | 8.89 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 371 B | 371 B | ✓ |
| Overall change | 371 B | 371 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| index.html gzip | 613 B | 613 B | ✓ |
| link.html gzip | 620 B | 620 B | ✓ |
| withRouter.html gzip | 608 B | 608 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| buildDuration | 14.3s | 15s | ⚠️ +790ms |
| nodeModulesSize | 46 MB | 46 MB | ⚠️ +13 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| 677f882d2ed8..e6c6.js gzip | 13.4 kB | 13.4 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 7.12 kB | 7.12 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 60.2 kB | 60.2 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| _app-8fbabfc..6440.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-d07ca..ef4d.js gzip | 3.71 kB | 3.71 kB | ✓ |
| amp-9716187d..0aa8.js gzip | 536 B | 536 B | ✓ |
| hooks-107e90..74c7.js gzip | 888 B | 888 B | ✓ |
| index-ac435c..ecf2.js gzip | 227 B | 227 B | ✓ |
| link-c31053f..c329.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..dc9d.js gzip | 303 B | 303 B | ✓ |
| withRouter-6..0e02.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.89 kB | 8.89 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 371 B | 371 B | ✓ |
| Overall change | 371 B | 371 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| _error.js | 1.37 MB | 1.37 MB | ⚠️ +8 B |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| 500.html | 2.65 kB | 2.65 kB | ✓ |
| amp.amp.html | 10.7 kB | 10.7 kB | ✓ |
| amp.html | 1.86 kB | 1.86 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1.37 MB | 1.37 MB | ⚠️ +8 B |
| link.js | 1.42 MB | 1.42 MB | ⚠️ +8 B |
| routerDirect.js | 1.42 MB | 1.42 MB | ⚠️ +8 B |
| withRouter.js | 1.42 MB | 1.42 MB | ⚠️ +8 B |
| Overall change | 7.01 MB | 7.01 MB | ⚠️ +40 B |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 5 Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| buildDuration | 15.5s | 14.7s | -756ms |
| nodeModulesSize | 46 MB | 46 MB | ⚠️ +13 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.746 | 2.539 | -0.21 |
| / avg req/sec | 910.3 | 984.56 | +74.26 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.547 | 1.516 | -0.03 |
| /error-in-render avg req/sec | 1615.6 | 1649.44 | +33.84 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| 597-3f457946..288d.js gzip | 13.3 kB | 13.3 kB | ✓ |
| 778-5cc798ac..5c25.js gzip | 7.05 kB | 7.05 kB | ✓ |
| framework.HASH.js gzip | 39.3 kB | 39.3 kB | ✓ |
| main-HASH.js gzip | 151 B | 151 B | ✓ |
| webpack-HASH.js gzip | 993 B | 993 B | ✓ |
| Overall change | 60.8 kB | 60.8 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| _app-5cc66b2..6f03.js gzip | 1.3 kB | 1.3 kB | ✓ |
| _error-55112..054d.js gzip | 3.68 kB | 3.68 kB | ✓ |
| amp-89a5460c..567f.js gzip | 558 B | 558 B | ✓ |
| hooks-8c2e74..be37.js gzip | 924 B | 924 B | ✓ |
| index-fec729..83b2.js gzip | 243 B | 243 B | ✓ |
| link-dd34d9b..0ade.js gzip | 1.66 kB | 1.66 kB | ✓ |
| routerDirect..5759.js gzip | 336 B | 336 B | ✓ |
| withRouter-1..98bf.js gzip | 334 B | 334 B | ✓ |
| Overall change | 9.03 kB | 9.03 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 349 B | 349 B | ✓ |
| Overall change | 349 B | 349 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| index.html gzip | 610 B | 610 B | ✓ |
| link.html gzip | 617 B | 617 B | ✓ |
| withRouter.html gzip | 605 B | 605 B | ✓ |
| Overall change | 1.83 kB | 1.83 kB | ✓ |

</details>

#### Diffs
<details>
<summary>Diff for <strong>index.html</strong></summary>

```diff
@@ -43,7 +43,7 @@
         "props": { "pageProps": {} },
         "page": "/",
         "query": {},
-        "buildId": "qeAPvgUfJuSwXoSupapcW",
+        "buildId": "QDYdDbA0iD-jlcWFzuedl",
         "isFallback": false,
         "gip": true
       }
@@ -77,11 +77,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/qeAPvgUfJuSwXoSupapcW/_buildManifest.js"
+      src="/_next/static/QDYdDbA0iD-jlcWFzuedl/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/qeAPvgUfJuSwXoSupapcW/_ssgManifest.js"
+      src="/_next/static/QDYdDbA0iD-jlcWFzuedl/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
<details>
<summary>Diff for <strong>link.html</strong></summary>

```diff
@@ -48,7 +48,7 @@
         "props": { "pageProps": {} },
         "page": "/link",
         "query": {},
-        "buildId": "qeAPvgUfJuSwXoSupapcW",
+        "buildId": "QDYdDbA0iD-jlcWFzuedl",
         "isFallback": false,
         "gip": true
       }
@@ -82,11 +82,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/qeAPvgUfJuSwXoSupapcW/_buildManifest.js"
+      src="/_next/static/QDYdDbA0iD-jlcWFzuedl/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/qeAPvgUfJuSwXoSupapcW/_ssgManifest.js"
+      src="/_next/static/QDYdDbA0iD-jlcWFzuedl/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
<details>
<summary>Diff for <strong>withRouter.html</strong></summary>

```diff
@@ -43,7 +43,7 @@
         "props": { "pageProps": {} },
         "page": "/withRouter",
         "query": {},
-        "buildId": "qeAPvgUfJuSwXoSupapcW",
+        "buildId": "QDYdDbA0iD-jlcWFzuedl",
         "isFallback": false,
         "gip": true
       }
@@ -77,11 +77,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/qeAPvgUfJuSwXoSupapcW/_buildManifest.js"
+      src="/_next/static/QDYdDbA0iD-jlcWFzuedl/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/qeAPvgUfJuSwXoSupapcW/_ssgManifest.js"
+      src="/_next/static/QDYdDbA0iD-jlcWFzuedl/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
</details>
Commit: d2ef4a6bfb3c37da26899c5a87ca914edd11d6aa

**Comment by ijjk** [comment]
## Failing test suites 
Commit: d2ef4a6bfb3c37da26899c5a87ca914edd11d6aa

**test/integration/i18n-support/test/index.test.js** 
- i18n Support > dev mode > should not redirect to accept-lang preferred locale with locale cookie
- i18n Support > production mode > should not redirect to accept-lang preferred locale with locale cookie
- i18n Support > serverless mode > should not redirect to accept-lang preferred locale with locale cookie

<details>
<summary>Expand output</summary>

  ● i18n Support › dev mode › should not redirect to accept-lang preferred locale with locale cookie

    expect(received).toBe(expected) // Object.is equality

    Expected: 200
    Received: 307

      1742 |     )
      1743 |
    > 1744 |     expect(res.status).toBe(200)
           |                        ^
      1745 |
      1746 |     const html = await res.text()
      1747 |     const $ = cheerio.load(html)

      at Object.<anonymous> (integration/i18n-support/test/shared.js:1744:24)
          at runMicrotasks (<anonymous>)

  ● i18n Support › production mode › should not redirect to accept-lang preferred locale with locale cookie

    expect(received).toBe(expected) // Object.is equality

    Expected: 200
    Received: 307

      1742 |     )
      1743 |
    > 1744 |     expect(res.status).toBe(200)
           |                        ^
      1745 |
      1746 |     const html = await res.text()
      1747 |     const $ = cheerio.load(html)

      at Object.<anonymous> (integration/i18n-support/test/shared.js:1744:24)
          at runMicrotasks (<anonymous>)

  ● i18n Support › serverless mode › should not redirect to accept-lang preferred locale with locale cookie

    expect(received).toBe(expected) // Object.is equality

    Expected: 200
    Received: 307

      1742 |     )
      1743 |
    > 1744 |     expect(res.status).toBe(200)
           |                        ^
      1745 |
      1746 |     const html = await res.text()
      1747 |     const $ = cheerio.load(html)

      at Object.<anonymous> (integration/i18n-support/test/shared.js:1744:24)
          at runMicrotasks (<anonymous>)

</details>

File: packages/next/next-server/lib/i18n/detect-locale-cookie.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { IncomingMessage } from 'http'
 
 export function detectLocaleCookie(req: IncomingMessage, locales: string[]) {
-  if (req.headers.cookie && req.headers.cookie.includes('NEXT_LOCALE')) {
+  if (req.headers.cookies && req.headers.cookies.hasOwnProperty('NEXT_LOCALE')) {
```
Comment (timneutkens) [inline packages/next/next-server/lib/i18n/detect-locale-cookie.ts:4]: `req.headers.cookies` is not a thing in `IncomingMessage`.

```suggestion
  if (req.headers.cookie && req.headers.cookie.hasOwnProperty('NEXT_LOCALE=')) {
```

**Review by timneutkens — CHANGES_REQUESTED** [review]
This is missing an integration test

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| buildDuration | 12.8s | 13.3s | ⚠️ +455ms |
| nodeModulesSize | 46 MB | 46 MB | ⚠️ +20 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.748 | 2.788 | ⚠️ +0.04 |
| / avg req/sec | 909.79 | 896.85 | ⚠️ -12.94 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.567 | 1.636 | ⚠️ +0.07 |
| /error-in-render avg req/sec | 1595.66 | 1528.14 | ⚠️ -67.52 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| 677f882d2ed8..e6c6.js gzip | 13.4 kB | 13.4 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 7.12 kB | 7.12 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 60.2 kB | 60.2 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| _app-8fbabfc..6440.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-d07ca..ef4d.js gzip | 3.71 kB | 3.71 kB | ✓ |
| amp-9716187d..0aa8.js gzip | 536 B | 536 B | ✓ |
| hooks-107e90..74c7.js gzip | 888 B | 888 B | ✓ |
| index-ac435c..ecf2.js gzip | 227 B | 227 B | ✓ |
| link-c31053f..c329.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..dc9d.js gzip | 303 B | 303 B | ✓ |
| withRouter-6..0e02.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.89 kB | 8.89 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 371 B | 371 B | ✓ |
| Overall change | 371 B | 371 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| index.html gzip | 613 B | 613 B | ✓ |
| link.html gzip | 620 B | 620 B | ✓ |
| withRouter.html gzip | 608 B | 608 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| buildDuration | 15.7s | 15.5s | -222ms |
| nodeModulesSize | 46 MB | 46 MB | ⚠️ +20 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| 677f882d2ed8..e6c6.js gzip | 13.4 kB | 13.4 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 7.12 kB | 7.12 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 60.2 kB | 60.2 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| _app-8fbabfc..6440.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-d07ca..ef4d.js gzip | 3.71 kB | 3.71 kB | ✓ |
| amp-9716187d..0aa8.js gzip | 536 B | 536 B | ✓ |
| hooks-107e90..74c7.js gzip | 888 B | 888 B | ✓ |
| index-ac435c..ecf2.js gzip | 227 B | 227 B | ✓ |
| link-c31053f..c329.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..dc9d.js gzip | 303 B | 303 B | ✓ |
| withRouter-6..0e02.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.89 kB | 8.89 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 371 B | 371 B | ✓ |
| Overall change | 371 B | 371 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| _error.js | 1.37 MB | 1.37 MB | ⚠️ +7 B |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| 500.html | 2.65 kB | 2.65 kB | ✓ |
| amp.amp.html | 10.7 kB | 10.7 kB | ✓ |
| amp.html | 1.86 kB | 1.86 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1.37 MB | 1.37 MB | ⚠️ +7 B |
| link.js | 1.42 MB | 1.42 MB | ⚠️ +7 B |
| routerDirect.js | 1.42 MB | 1.42 MB | ⚠️ +7 B |
| withRouter.js | 1.42 MB | 1.42 MB | ⚠️ +7 B |
| Overall change | 7.01 MB | 7.01 MB | ⚠️ +35 B |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 5 Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| buildDuration | 15.8s | 16.1s | ⚠️ +322ms |
| nodeModulesSize | 46 MB | 46 MB | ⚠️ +20 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.934 | 2.829 | -0.1 |
| / avg req/sec | 852.09 | 883.7 | +31.61 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.679 | 1.647 | -0.03 |
| /error-in-render avg req/sec | 1488.83 | 1517.84 | +29.01 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| 597-3f457946..288d.js gzip | 13.3 kB | 13.3 kB | ✓ |
| 778-5cc798ac..5c25.js gzip | 7.05 kB | 7.05 kB | ✓ |
| framework.HASH.js gzip | 39.3 kB | 39.3 kB | ✓ |
| main-HASH.js gzip | 151 B | 151 B | ✓ |
| webpack-HASH.js gzip | 993 B | 993 B | ✓ |
| Overall change | 60.8 kB | 60.8 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| _app-5cc66b2..6f03.js gzip | 1.3 kB | 1.3 kB | ✓ |
| _error-55112..054d.js gzip | 3.68 kB | 3.68 kB | ✓ |
| amp-89a5460c..567f.js gzip | 558 B | 558 B | ✓ |
| hooks-8c2e74..be37.js gzip | 924 B | 924 B | ✓ |
| index-fec729..83b2.js gzip | 243 B | 243 B | ✓ |
| link-dd34d9b..0ade.js gzip | 1.66 kB | 1.66 kB | ✓ |
| routerDirect..5759.js gzip | 336 B | 336 B | ✓ |
| withRouter-1..98bf.js gzip | 334 B | 334 B | ✓ |
| Overall change | 9.03 kB | 9.03 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 349 B | 349 B | ✓ |
| Overall change | 349 B | 349 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | timbset/next.js patch-1 | Change |
| - | - | - | - |
| index.html gzip | 610 B | 610 B | ✓ |
| link.html gzip | 617 B | 617 B | ✓ |
| withRouter.html gzip | 605 B | 605 B | ✓ |
| Overall change | 1.83 kB | 1.83 kB | ✓ |

</details>

#### Diffs
<details>
<summary>Diff for <strong>index.html</strong></summary>

```diff
@@ -43,7 +43,7 @@
         "props": { "pageProps": {} },
         "page": "/",
         "query": {},
-        "buildId": "MJpGRY2DIZUuwCoLR69lB",
+        "buildId": "Wd_FNz5EbEXmaLSBz7h8I",
         "isFallback": false,
         "gip": true
       }
@@ -77,11 +77,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/MJpGRY2DIZUuwCoLR69lB/_buildManifest.js"
+      src="/_next/static/Wd_FNz5EbEXmaLSBz7h8I/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/MJpGRY2DIZUuwCoLR69lB/_ssgManifest.js"
+      src="/_next/static/Wd_FNz5EbEXmaLSBz7h8I/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
<details>
<summary>Diff for <strong>link.html</strong></summary>

```diff
@@ -48,7 +48,7 @@
         "props": { "pageProps": {} },
         "page": "/link",
         "query": {},
-        "buildId": "MJpGRY2DIZUuwCoLR69lB",
+        "buildId": "Wd_FNz5EbEXmaLSBz7h8I",
         "isFallback": false,
         "gip": true
       }
@@ -82,11 +82,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/MJpGRY2DIZUuwCoLR69lB/_buildManifest.js"
+      src="/_next/static/Wd_FNz5EbEXmaLSBz7h8I/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/MJpGRY2DIZUuwCoLR69lB/_ssgManifest.js"
+      src="/_next/static/Wd_FNz5EbEXmaLSBz7h8I/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
<details>
<summary>Diff for <strong>withRouter.html</strong></summary>

```diff
@@ -43,7 +43,7 @@
         "props": { "pageProps": {} },
         "page": "/withRouter",
         "query": {},
-        "buildId": "MJpGRY2DIZUuwCoLR69lB",
+        "buildId": "Wd_FNz5EbEXmaLSBz7h8I",
         "isFallback": false,
         "gip": true
       }
@@ -77,11 +77,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/MJpGRY2DIZUuwCoLR69lB/_buildManifest.js"
+      src="/_next/static/Wd_FNz5EbEXmaLSBz7h8I/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/MJpGRY2DIZUuwCoLR69lB/_ssgManifest.js"
+      src="/_next/static/Wd_FNz5EbEXmaLSBz7h8I/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
</details>
Commit: 0e92463a25d772d30ee1f1a2fe8b50c397bf32b8

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 0e92463a25d772d30ee1f1a2fe8b50c397bf32b8

**test/integration/i18n-support-base-path/test/index.test.js** 
- i18n Support basePath > dev mode > should not redirect to accept-lang preferred locale with locale cookie
- i18n Support basePath > production mode > should not redirect to accept-lang preferred locale with locale cookie
- i18n Support basePath > serverless mode > should not redirect to accept-lang preferred locale with locale cookie

<details>
<summary>Expand output</summary>

  ● i18n Support basePath › dev mode › should not redirect to accept-lang preferred locale with locale cookie

    expect(received).toBe(expected) // Object.is equality

    Expected: 200
    Received: 307

      1742 |     )
      1743 |
    > 1744 |     expect(res.status).toBe(200)
           |                        ^
      1745 |
      1746 |     const html = await res.text()
      1747 |     const $ = cheerio.load(html)

      at Object.<anonymous> (integration/i18n-support/test/shared.js:1744:24)
          at runMicrotasks (<anonymous>)

  ● i18n Support basePath › production mode › should not redirect to accept-lang preferred locale with locale cookie

    expect(received).toBe(expected) // Object.is equality

    Expected: 200
    Received: 307

      1742 |     )
      1743 |
    > 1744 |     expect(res.status).toBe(200)
           |                        ^
      1745 |
      1746 |     const html = await res.text()
      1747 |     const $ = cheerio.load(html)

      at Object.<anonymous> (integration/i18n-support/test/shared.js:1744:24)
          at runMicrotasks (<anonymous>)

  ● i18n Support basePath › serverless mode › should not redirect to accept-lang preferred locale with locale cookie

    expect(received).toBe(expected) // Object.is equality

    Expected: 200
    Received: 307

      1742 |     )
      1743 |
    > 1744 |     expect(res.status).toBe(200)
           |                        ^
      1745 |
      1746 |     const html = await res.text()
      1747 |     const $ = cheerio.load(html)

      at Object.<anonymous> (integration/i18n-support/test/shared.js:1744:24)
          at runMicrotasks (<anonymous>)

</details>

**Pull request closed without merging**